custom_code/templatetags/custom_code_tags.py

```diff
@@ -865,28 +861,27 @@ def observation_summary(context, target=None, time='previous'):
     """
     if target:
         if settings.TARGET_PERMISSIONS_ONLY:
-            observations = target.observationrecord_set.all()
+            observations = target.observationrecord_set.all().prefetch_related('observationgroup_set', 'observationgroup_set__dynamiccadence_set')
         else:
             observations = get_objects_for_user(
                                 context['request'].user,
                                 'tom_observations.view_observationrecord',
-                                ).filter(target=target)
+                                ).filter(target=target).prefetch_related('observationgroup_set', 'observationgroup_set__dynamiccadence_set')
     else:
-        observations = ObservationRecord.objects.all()
+        observations = ObservationRecord.objects.all().prefetch_related('observationgroup_set', 'observationgroup_set__dynamiccadence_set')
 
     observations = observations.order_by('parameters__start')
 
     if time == 'ongoing':
-        cadences = DynamicCadence.objects.filter(active=True, observation_group__in=ObservationGroup.objects.filter(name__in=[o.parameters.get('name', '') for o in observations]))
-    
+        observation_groups = {obsgroup for observation in observations for obsgroup in observation.observationgroup_set.all() for cadence in obsgroup.dynamiccadence_set.all() if cadence.active}
     else:
         if time == 'pending':
             observations = observations.filter(observation_id='template pending')
-        cadences = DynamicCadence.objects.filter(active=False, observation_group__in=ObservationGroup.objects.filter(name__in=[o.parameters.get('name', '') for o in observations]))
-    
+        observation_groups = {obsgroup for observation in observations for obsgroup in observation.observationgroup_set.all() for cadence in obsgroup.dynamiccadence_set.all() if not cadence.active}
```

Comment:
Cadences weren't used other than to get their observation_groups so I just changed the code to get those directly and remove the cadences reference. This function still does a bunch of queries but its at least a little less than before.